.github/workflows/milestone-tracking.yml

```diff
@@ -0,0 +1,41 @@
+name: Milestone tracking
+on:
+  pull_request:
+    types: [closed]
+jobs:
+  add_milestone:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'dotnet/roslyn' && github.event.pull_request.merged_at != null && github.event.pull_request.milestone == null && (github.event.pull_request.base.ref == 'main' || github.event.pull_request.base.ref == 'main-vs-deps') }}
+    steps:
+    - name: Get milestone data
+      env:
+        ORGANIZATION: dotnet
+        REPOSITORY: roslyn-analyzers
+        MILESTONE_NAME: vNext
+      run: |
+        gh api graphql -f query='
+          query($org: String!, $repo: String!, $milestone: String!) {
+            repository(name: $repo, owner: $org) {
+              milestones(query: $milestone, first: 2) {
+                nodes {
+                  id
+                  title
+                }
+              }
+            }
+          }' -f org=$ORGANIZATION -f repo=$REPOSITORY -f milestone="$MILESTONE_NAME" > milestone_data.json
```

Comment:
where is `MILESTONE_NAME` going to be defined?

Reply:
Line 14 above